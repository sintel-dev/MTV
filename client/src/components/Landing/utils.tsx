export const colorSchemes = {
  tag: [
    '#FFCD00', // investigate
    '#7CA8FF', // do not investigate
    '#A042FF', // postpone
    '#F64242', // problem
    '#F5FF00', // previously seen
    '#45F642', // normal
    '#C7C7C7', // untagged
  ],
};

// @TODO - later investigation for refactoring (fromIDToTag/fromTagToId)
export const tagSeq = [
  'Investigate',
  'Do not Investigate',
  'Postpone',
  'Problem',
  'Previously seen',
  'Normal',
  'Untagged',
];
const tagClassNames = ['investigate', 'not_investigate', 'postpone', 'problem', 'seen', 'normal', 'untagged'];

export const fromTagToClassName = (tag: string): string => {
  switch (tag) {
    case tagSeq[0]:
      return tagClassNames[0];
    case tagSeq[1]:
      return tagClassNames[1];
    case tagSeq[2]:
      return tagClassNames[2];
    case tagSeq[3]:
      return tagClassNames[3];
    case tagSeq[4]:
      return tagClassNames[4];
    case tagSeq[5]:
      return tagClassNames[5];
    default:
      return 'untagged';
  }
};

export const fromIDtoTag = (id: string): string => {
  switch (id) {
    case '1':
      return tagSeq[0];
    case '2':
      return tagSeq[1];
    case '3':
      return tagSeq[2];
    case '4':
      return tagSeq[3];
    case '5':
      return tagSeq[4];
    case '6':
      return tagSeq[5];
    default:
      return 'untagged';
  }
};

export const fromTagToID = (tag: string): string => {
  switch (tag) {
    case tagSeq[0]:
      return '1';
    case tagSeq[1]:
      return '2';
    case tagSeq[2]:
      return '3';
    case tagSeq[3]:
      return '4';
    case tagSeq[4]:
      return '5';
    case tagSeq[5]:
      return '6';
    default:
      return 'untagged';
  }
};

export const getTagColor = (tag: string): string => {
  let colorIdx: number | undefined;
  for (let i = 0; i < tagSeq.length; i += 1) {
    if (tagSeq[i] === tag) {
      colorIdx = i;
    }
  }
  if (colorIdx === undefined) {
    colorIdx = 6;
  }
  return colorSchemes.tag[colorIdx];
};
