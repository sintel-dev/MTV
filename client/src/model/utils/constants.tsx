export const API_URL = 'http://127.0.0.1:3000/api/v1/';
export const GOOGLE_CLIEN_ID = '796048243680-341g672umjji4m4okfgppm2ogfkrh8mj.apps.googleusercontent.com';
export const SESSION_TOKEN = 'AUTHORIZATION';
