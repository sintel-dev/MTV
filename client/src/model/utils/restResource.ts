import { AxiosRequestConfig, AxiosInstance } from 'axios';

export default class Resource<T, R> {
  /**
   *
   * @param server AxiosInstance
   * @param url Recourse end point
   */
  constructor(private server: AxiosInstance, private url: string) {}

  /**
   * Get all items
   * @param data body data - in most cases, this should be {}
   * @param params URL parameters
   */
  public all(data = {}, params = {}): Promise<R> {
    const promise = this.server.get<R>(this.url, { data, params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Get one item
   * @param id the item ID (name) to be retrieved
   * @param data body data â€“ in most cases, this should be {}
   * @param params URL parameters
   */
  public find(id, data = {}, params = {}): Promise<T> {
    const newUrl = `${this.url}${id}`;
    const promise = this.server.get<T>(newUrl, { data, params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Create an item.
   * @param data the item data to create
   * @param params URL parameters
   */
  public create(data = {}, params = {}): Promise<T> {
    const promise = this.server.post<T>(this.url, data, { params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Delete an item.
   * @param id the item ID (name) to delete
   * @returns Promise of the response's status
   */
  public delete(id): Promise<number> {
    const newUrl = `${this.url}${id}/`;
    const promise = this.server.delete<number>(newUrl);
    return promise
      .then((res) => res.status)
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Update an item.
   * @param id the item ID (name) to update
   * @param data the updated data of this item
   * @param params URL parameters
   */
  public update(id, data = {}, params = {}): Promise<T> {
    const newUrl = `${this.url}${id}/`;
    const promise = this.server.put<T>(newUrl, data, { params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Allow customize your request.
   * @param config request config
   * @param id item id
   */
  public request<Z>(config: AxiosRequestConfig = {}, id?: string): Promise<Z> {
    let newUrl = this.url;
    if (id !== undefined) {
      newUrl += `${id}/`;
    }
    const promise = this.server.get<Z>(newUrl, config);
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }
}
