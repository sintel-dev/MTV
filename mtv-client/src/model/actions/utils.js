const baseUrl = '/api/v1/';

export const api = {
    get(url, dataType = 'json') {
        const apiUrl = `${baseUrl}${url}`;
        const promise = fetch(apiUrl, {
            mode: 'cors',
            headers: {
                'Access-Control-Allow-Origin': '*',
            },
        });
        return promise.then(response => {
            if (response.status !== 204) {
                return dataType !== 'json' ? response.text : response.json();
            }
            return null;
        })
        .catch(err => console.error(err));
    },

    // post(url, data) {
    //     const apiUrl = `${baseUrl}${url}`;
    //     return fetch(apiUrl, {
    //         mode: 'cors',
    //         method: 'POST',
    //         headers: {
    //             'Access-Control-Allow-Origin': '*',
    //             'Content-Type': 'application/json',
    //             // 'Authorization': lscache.get('authorization')
    //         },
    //         body: JSON.stringify(data)
    //     });
    // }
};
