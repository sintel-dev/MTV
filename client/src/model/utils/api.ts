import axios, { AxiosRequestConfig, AxiosInstance } from 'axios';
import Cookies from 'js-cookie';
import { API_URL, SESSION_TOKEN } from './constants';
// import { SESSION_TOKEN } from '@nx-react/session';

import Resource from './restResource';
import {
  DatasetDataType,
  DatasetsResponse,
  DatarunDataType,
  DatarunsResponse,
  ExperimentDataType,
  ExperimentsResponse,
  PipelinesResponse,
  EventsResponse,
  CommentsResponse,
  DataResponse,
} from '../types/index';
import { PipelineDataType } from '../types/pipeline';
import { EventDataType } from '../types/event';
import { CommentDataType } from '../types/comment';

export class RestClient {
  private server: AxiosInstance;

  // default config for http request
  private requestConfig: AxiosRequestConfig = {
    baseURL: API_URL,
    // timeout: 2000,
  };

  public test: Resource<any, any>;

  public datasets: Resource<DatasetDataType, DatasetsResponse>;

  public dataruns: Resource<DatarunDataType, DatarunsResponse>;

  public experiments: Resource<ExperimentDataType, ExperimentsResponse>;

  public pipelines: Resource<PipelineDataType, PipelinesResponse>;

  public events: Resource<EventDataType, EventsResponse>;

  public comments: Resource<CommentDataType, CommentsResponse>;

  public data: Resource<any, DataResponse>;

  /**
   *
   * @param config AxiosRequestConfig
   */
  constructor(config: AxiosRequestConfig) {
    this.requestConfig = { ...this.requestConfig, ...config };
    this.server = axios.create(this.requestConfig);

    // add resources
    this.test = new Resource(this.server, 'test/');
    this.datasets = new Resource(this.server, 'datasets/');
    this.dataruns = new Resource(this.server, 'dataruns/');
    this.experiments = new Resource(this.server, 'experiments/');
    this.pipelines = new Resource(this.server, 'pipelines/');
    this.events = new Resource(this.server, 'events/');
    this.comments = new Resource(this.server, 'comments/');
    this.data = new Resource(this.server, 'data/');
    this.server.interceptors.request.use(this.requestInterceptor);
    this.server.interceptors.response.use(this.responseSuccessInterceptor, this.responseFailInterceptor);
  }

  public requestInterceptor(config) {
    const authHeader = Cookies.get(SESSION_TOKEN);
    if (!authHeader) {
      return config;
    }

    return {
      ...config,
      headers: {
        ...config.headers,
        Authorization: authHeader,
      },
    };
  }

  public responseSuccessInterceptor(response) {
    return response;
  }

  public responseFailInterceptor(error) {
    if (error.response.status === 401) {
      window.location.href = '/logout';
    }

    return Promise.reject(error);
  }
}

export default new RestClient({
  baseURL: 'http://127.0.0.1:3000/api/v1/',
});
