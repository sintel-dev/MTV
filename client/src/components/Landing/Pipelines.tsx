import React from 'react';
import { connect } from 'react-redux';
import Loader from '../Common/Loader';
import { getPipelinesData, getSelectedPipeline } from '../../model/selectors/projects';
import { selectPipeline } from '../../model/actions/landing';
import { RootState, PipelineDataType } from '../../model/types';

type StateProps = ReturnType<typeof mapState>;
type DispatchProps = ReturnType<typeof mapDispatch>;
type Props = StateProps & DispatchProps;

const Pipelines: React.FC<Props> = props => {
  const { onSelectPipeline, selectedPipeline, pipelinesData } = props;
  const { pipelineList, isPipelinesLoading } = pipelinesData;

  return (
    <div className="item-row scroll-style" id="pipelines">
      <h2>Pipelines</h2>
      <div className="item-wrapper">
        <Loader isLoading={isPipelinesLoading}>
          {pipelineList.length ? (
            pipelineList.map((pipeline, index) =>
              renderPipeline({ pipeline, index, onSelectPipeline, selectedPipeline }),
            )
          ) : (
            <p>No pipelines have been found</p>
          )}
        </Loader>
      </div>
    </div>
  );
};

let props: Props;
type renderPipelineProps = {
  pipeline: PipelineDataType;
  index: number;
  onSelectPipeline: typeof props.onSelectPipeline;
  selectedPipeline: typeof props.selectedPipeline;
};

const renderPipeline: React.FC<renderPipelineProps> = ({ pipeline, index, onSelectPipeline, selectedPipeline }) => {
  const activeClass = selectedPipeline === pipeline.name ? 'active' : '';

  return (
    <div className={`cell ${activeClass}`} key={index} onClick={() => onSelectPipeline(pipeline.name)}>
      <h3>{pipeline.name}</h3>
      <div className="item-data">
        <ul>
          <li>DC: {pipeline.insert_time.substring(0, 10)}</li>
          <li>By: {pipeline.created_by || 'null'}</li>
        </ul>
      </div>
    </div>
  );
};

const mapState = (state: RootState) => ({
  pipelinesData: getPipelinesData(state),
  selectedPipeline: getSelectedPipeline(state),
});

const mapDispatch = (dispatch: Function) => ({
  onSelectPipeline: pipelineName => dispatch(selectPipeline(pipelineName)),
});

export default connect<StateProps, DispatchProps, {}, RootState>(mapState, mapDispatch)(Pipelines);
