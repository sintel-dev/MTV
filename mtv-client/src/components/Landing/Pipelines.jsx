import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Loader from '../Common/Loader';
import {
  getPipelinesData,
  getSelectedPipeline,
} from '../../model/selectors/projects';
import { selectPipeline } from '../../model/actions/landing';

const renderPipeline = (pipeline, index, onPipelineSelect, selectedPipeline) => {
  const activeClass = selectedPipeline === pipeline.name ? 'active' : '';

  return (
    <div className={`cell ${activeClass}`} key={index} onClick={() => onPipelineSelect(pipeline.name)}>
      <h3>{pipeline.name}</h3>
      <div className="item-data">
        <ul>
          <li>DC: {pipeline.insert_time.substring(0, 10)}</li>
          <li>By: {pipeline.created_by || 'null' }</li>
        </ul>
      </div>
    </div>
  );
};

const Pipelines = (props) => {
  const { pipelineList, isPipelinesLoading } = props.pipelinesData;
  const { onSelectPipeline, selectedPipeline } = props;

  return (
    <div className="item-row scroll-style" id="pipelines">
      <h2>Pipelines</h2>
      <div className="item-wrapper">
        <Loader isLoading={isPipelinesLoading}>
          {
            pipelineList.length ?
              pipelineList.map((pipeline, index) =>
              renderPipeline(pipeline, index, onSelectPipeline, selectedPipeline)) :
              <p>No pipelines have been found</p>
          }
        </Loader>
      </div>
    </div>
  );
};

Pipelines.propTypes = {
  pipelinesData: PropTypes.object,
  onSelectPipeline: PropTypes.func,
  selectedPipeline: PropTypes.string,
};

export default connect(state => ({
  pipelinesData: getPipelinesData(state),
  selectedPipeline: getSelectedPipeline(state),
}), dispatch => ({
  onSelectPipeline: (pipelineName) => dispatch(selectPipeline(pipelineName)),
}))(Pipelines);
