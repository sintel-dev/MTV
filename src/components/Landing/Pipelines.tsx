import React from 'react';
import { connect } from 'react-redux';
import { getPipelinesData, getSelectedPipeline } from 'src/model/selectors/projects';
import { selectPipeline } from 'src/model/actions/landing';
import { RootState, PipelineDataType } from 'src/model/types';
import Loader from '../Common/Loader';

type StateProps = ReturnType<typeof mapState>;
type DispatchProps = ReturnType<typeof mapDispatch>;
type Props = StateProps & DispatchProps;

const Pipelines: React.FC<Props> = (props) => {
  const { onSelectPipeline, selectedPipeline, pipelinesData } = props;
  const { pipelineList, isPipelinesLoading } = pipelinesData;

  return (
    <div className="item-row scroll-style" id="pipelines">
      <h2>Pipelines</h2>
      <div className="item-wrapper">
        <Loader isLoading={isPipelinesLoading}>
          {pipelineList.length ? (
            pipelineList.map((pipeline, index) =>
              RenderPipeline({ pipeline, index, onSelectPipeline, selectedPipeline }),
            )
          ) : (
            <p>No pipelines have been found</p>
          )}
        </Loader>
      </div>
    </div>
  );
};

let props: Props;
type PipelineProps = {
  pipeline: PipelineDataType;
  index: number;
  onSelectPipeline: typeof props.onSelectPipeline;
  selectedPipeline: typeof props.selectedPipeline;
};

export const RenderPipeline: React.FC<PipelineProps> = ({
  pipeline,
  index,
  onSelectPipeline,
  selectedPipeline,
}: PipelineProps) => {
  const activeClass: string = selectedPipeline === pipeline.name ? 'active' : '';
  const { name, insert_time, created_by } = pipeline;

  return (
    <div className={`cell ${activeClass}`} key={index} onClick={() => onSelectPipeline(name)}>
      <h3>{name}</h3>
      <div className="item-data">
        <ul>
          <li>DC: {insert_time.substring(0, 10)}</li>
          <li>By: {created_by || 'null'}</li>
        </ul>
      </div>
    </div>
  );
};

const mapState = (state: RootState) => ({
  pipelinesData: getPipelinesData(state),
  selectedPipeline: getSelectedPipeline(state),
});

const mapDispatch = (dispatch: Function) => ({
  onSelectPipeline: (pipelineName: string) => dispatch(selectPipeline(pipelineName)),
});

export default connect<StateProps, DispatchProps, {}, RootState>(mapState, mapDispatch)(Pipelines);
