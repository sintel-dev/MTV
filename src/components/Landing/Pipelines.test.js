import React from 'react';
import { RenderPipeline } from './Pipelines';
import { renderWithStore, TestWrapper } from '../../tests/utils';

describe('Pipeline test', () => {
  it('Should render pipeline without crashing', () => {
    const pipelinesProps = {
      pipeline: {
        id: '5da7cc308c5ceceb9f28901c',
        insert_time: '2019-10-17T02:04:32.151000',
        name: 'lstm',
        created_by: null,
      },
      index: 0,
    };
    const mountedPipeline = renderWithStore(
      {},
      <TestWrapper>
        <RenderPipeline {...pipelinesProps} />
      </TestWrapper>,
    );
    expect(mountedPipeline).toMatchSnapshot();
  });
});
