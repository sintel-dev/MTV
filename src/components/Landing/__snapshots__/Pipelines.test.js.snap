// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipeline test Should render pipeline without crashing 1`] = `
<div
  class="cell "
>
  <h3>
    lstm
  </h3>
  <div
    class="item-data"
  >
    <ul>
      <li>
        DC: 2019-10-17
      </li>
      <li>
        By: null
      </li>
    </ul>
  </div>
</div>
`;
